frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx

```diff
@@ -71,15 +73,53 @@ const TopologyFilterBar: React.FC<TopologyFilterBarProps> = ({
   const kialiLink = getNamespaceDashboardKialiLink(consoleLinks, namespace);
   const queryParams = useQueryParams();
   const searchQuery = queryParams.get('searchQuery') || '';
+  const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(false);
 
   const onTextFilterChange = (text) => {
     const query = text?.trim();
     onSearchChange(query);
   };
 
+  React.useEffect(() => {
+    const onKeyDown = (e: KeyboardEvent) => {
+      e.preventDefault();
+      const { nodeName } = e.target as Element;
+      if (nodeName === 'INPUT' || nodeName === 'TEXTAREA') {
+        return;
+      }
+      switch (e.code) {
+        case 'KeyF':
+          e.ctrlKey && setIsQuickSearchActive(true);
+          break;
+        case 'Escape':
+          setIsQuickSearchActive(false);
```

Comment:
The closing of the modal will have to be controlled within the modal because pressing the esc key will first clear the text. Only when the text is clear will pressing esc close the modal.

Reply:
Yeah I know but just wanted to keep this working on its own as this PR will go in first, but I can remove this.